pkg/steps/multi_stage.go

```diff
@@ -562,7 +568,14 @@ func (s *multiStageTestStep) runPod(ctx context.Context, pod *coreapi.Pod, notif
 		if s.config.Metadata.Variant != "" {
 			linksText.WriteString(fmt.Sprintf("&variant=%s", s.config.Metadata.Variant))
 		}
-		return fmt.Errorf("%q pod %q failed: %w\n%s", s.name, pod.Name, err, linksText.String())
+		status := "failed"
+		if pod != nil && pod.Status.Phase == coreapi.PodFailed && pod.Status.Reason == "DeadlineExceeded" {
```

Comment:
Isn't `DeadlineExceeded` a const somewhere?

Reply:
Not in k8s api, iiuc its being set by kubelet, which is not vendored